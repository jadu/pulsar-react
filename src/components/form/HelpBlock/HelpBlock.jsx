import React from 'react';

export default class HelpBlock extends React.Component {
  render() {
    if (!this.props.children) {
      return null;
    }

    let {
      children,
      helpGuid,
      ...props} = this.props;

    return (
      <span id={helpGuid} className="help-block">{children}</span>
    );
  }
}