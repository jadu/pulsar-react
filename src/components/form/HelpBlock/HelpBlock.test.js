'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import HelpBlock from './HelpBlock';

it('Is null if no Help is provided', () => {
  const tree = render(
    <HelpBlock></HelpBlock>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the help block when help is provided', () => {
  const tree = render(
    <HelpBlock>Foo</HelpBlock>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the help block with helpGuid as the ID', () => {
  const tree = render(
    <HelpBlock helpGuid="bar">Foo</HelpBlock>
  );
  expect(tree).toMatchSnapshot();
});


