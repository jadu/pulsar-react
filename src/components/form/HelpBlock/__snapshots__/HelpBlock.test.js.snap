// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is null if no Help is provided 1`] = `null`;

exports[`Renders the help block when help is provided 1`] = `
<span
  class="help-block"
>
  Foo
</span>
`;

exports[`Renders the help block with helpGuid as the ID 1`] = `
<span
  class="help-block"
  id="bar"
>
  Foo
</span>
`;
